File: Pinball App Android/app/src/main/java/com/pillyliu/pinprofandroid/library/LibraryImportedSourceModels.kt
package com.pillyliu.pinprofandroid.library

internal enum class ImportedSourceProvider(val rawValue: String) {
    OPDB("opdb"),
    PINBALL_MAP("pinball_map"),
    MATCH_PLAY("match_play");

    companion object {
        fun fromRaw(raw: String?): ImportedSourceProvider? = entries.firstOrNull { it.rawValue == raw }
    }
}

internal data class ImportedSourceRecord(
    val id: String,
    val name: String,
    val type: LibrarySourceType,
    val provider: ImportedSourceProvider,
    val providerSourceId: String,
    val machineIds: List<String>,
    val lastSyncedAtMs: Long? = null,
    val searchQuery: String? = null,
    val distanceMiles: Int? = null,
)

File: Pinball App 2/Pinball App 2/library/LibraryImportedSourceModels.swift
import Foundation

enum PinballImportedSourceProvider: String, Codable {
    case opdb
    case pinballMap = "pinball_map"
    case matchPlay = "match_play"
}

struct PinballImportedSourceRecord: Codable, Identifiable, Hashable {
    let id: String
    var name: String
    var type: PinballLibrarySourceType
    var provider: PinballImportedSourceProvider
    var providerSourceID: String
    var machineIDs: [String]
    var lastSyncedAt: Date?
    var searchQuery: String?
    var distanceMiles: Int?
}

func inferredImportedSourceProvider(type: PinballLibrarySourceType, id: String) -> PinballImportedSourceProvider {
    switch type {
    case .manufacturer:
        return .opdb
    case .tournament:
        return .matchPlay
    case .venue:
        return id.hasPrefix("venue--pm-") ? .pinballMap : .opdb
    case .category:
        return .opdb
    }
}
